include/net/golioth.h

```diff
@@ -7,6 +7,9 @@
 #ifndef GOLIOTH_INCLUDE_NET_GOLIOTH_H_
 #define GOLIOTH_INCLUDE_NET_GOLIOTH_H_
 
+#if defined(CONFIG_GOLIOTH_RPC)
+#include <net/golioth/rpc.h>
+#endif
```

Comment:
Would prefer to always include headers, regardless of CONFIG_ options. This is the policy followed by Zephyr. The advantage of that is the usage of `IS_ENABLED()`:
```
if (IS_ENABLED(CONFIG_GOLIOTH_RPC)) {
    golioth_rpc_register();
}
```
in application code, as opposed to:
```
#if CONFIG_GOLIOTH_RPC)
    golioth_rpc_register();
#endif
```
The advantage is that it is C compiler that removes unused code, not C preprocessor. This makes code more maintainable and harder to introduce errors based on option being enabled or disabled.

Reply:
:+1: I'll remove the preprocessor statements, and condition things with `IS_ENABLED`

To avoid the extra storage in `struct golioth_client` in case RPC is not enabled, what do you think about doing this in `golioth_rpc.h`?

```c
struct golioth_rpc {
#if defined(CONFIG_GOLIOTH_RPC)
	bool initialized;
	struct coap_reply observe_reply;
	struct golioth_rpc_method methods[CONFIG_GOLIOTH_RPC_MAX_NUM_METHODS];
	int num_methods;
#endif
};
```

Reply:
Seems fine to me!